File: day1.js
let {sample, input} = require('./input1.js');

function parse(input) {
	let rows = input.trim().split("\n");
	let elves = [];
	let current = {items: [], total: 0};

	for (let i=0; i < rows.length; i++) {
		if (rows[i].length) {
			let amt = parseInt(rows[i]);
			current.items.push(amt);
			current.total += amt;
		} else {
			elves.push(current);
			current = {items: [], total: 0};
		}
	}
	if (current.total) elves.push(current);

	return elves;
}

const numSort = (a, b) => a - b;
const sum = (a, b) => a + b;

let data = parse(sample);
data.sort((a, b) => a.total - b.total);
console.log(data);
let totals = data.map(elf => elf.total);
let result = totals.slice(-3);

//totals.sort(numSort);

console.log(result);
console.log(result.reduce(sum, 0));

File: day1.exs
defmodule Aoc do
	# Breaks input into groups. Returns list of groups, where each group is a list of integers
	def parseGroups(input) do
		input
		|> String.split("\n\n") # Breaks into groups where separated by blank lines
		|> Enum.map(fn g -> (String.split(g) |> Enum.map(&String.to_integer/1)) end) # Splits groups on newlines and parses to integers
	end

	def part1(input) do
		parseGroups(input)
		|> Enum.map(&Enum.sum/1)
		|> Enum.max
	end

	def part2(input) do
		parseGroups(input)
		|> Enum.map(&Enum.sum/1) # Total each group
		|> Enum.sort(:desc)
		|> Enum.slice(1..3)
		|> Enum.sum
	end

	def run(inputFile) do
		IO.puts(["Reading ", inputFile])
		{:ok, input} = File.read(inputFile)
		IO.inspect ["Part 1:", part1(input)]
		IO.inspect ["Part 2:", part2(input)]
	end
end

Aoc.run("sample.1")
Aoc.run("input.1")
